sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -427,32 +431,49 @@ public void onSubscribe(Subscription s) {
 
             @Override
             public void onNext(ByteBuffer bytes) {
-                isWriting = true;
-                outFile.write(bytes, pos, null, onWriteCompleted);
+                if (isWriting) {
+                    if (!writeQueue.offer(bytes)) {
+                        onError(new IllegalStateException("Dropped onNext when writing to file."));
+                    }
+                } else {
+                    isWriting = true;
+
+                    try {
+                        outFile.write(bytes, pos, null, onWriteCompleted);
+                    } catch (Throwable throwable) {
+                        onError(throwable);
+                    }
+                }
             }
 
 
             final CompletionHandler<Integer, Object> onWriteCompleted = new CompletionHandler<Integer, Object>() {
```

Comment:
Also, we should use `ByteBuffer` instead of `Object`.

Reply:
Updated